client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java

```diff
@@ -317,6 +356,32 @@ private void notifyListeners(Latest<T> latest) {
         }
     }
 
+    private void updateLatestCommitAsync(Revision revision) {
+        if (meterRegistry == null) {
+            return;
+        }
+
+        centralDogma.forRepo(projectName, repositoryName)
+                    .history()
+                    .get(revision, revision)
+                    .whenComplete((commits, e) -> {
+                        if (e == null) {
+                            commits.stream().findFirst().ifPresent(commit -> {
+                                // noinspection ConstantValue
+                                if (latestCommit.getAndSet(commit) == null) {
+                                    // emit metrics once the values are ready
+                                    meterRegistry.gauge(LATEST_COMMIT_TIME_METER_NAME, tags, latestCommit,
```

Comment:
Question) I understand the usefulness of last commit time since it may be more convenient to check when a file was last committed as opposed to checking the revision numbers.

Having said this, this commit time may not accurate represent the current watched object since it is done in a separate IO.

What do you think of handling commit time separately by embedding commit time information in the watch response? This will involve server-side changes as well, so it may be better to handle in a separate PR.

Reply:
> What do you think of handling commit time separately by embedding commit time information in the watch response? This will involve server-side changes as well, so it may be better to handle in a separate PR.

Yes, I actually wanted this as well. However, since the API interface on the **Central Dogma server** side will also change, there seem to be many things that need to be discussed.

Reply:
Isn’t it more important when the client receives a watch for a revision? 
If you think the commit time recorded in the server is meaningful, I prefer @jrhee17's approach.

Reply:
Ahhh, now that you mention it, that makes sense.

It seems better to simply record the value at the time the watcher receives it, without calling the get history API.
I'll make that change. It should be much simpler.

e.g. `centraldogma.client.watcher.latest.received.time`